### PR TITLE
implement function instr

### PR DESCRIPTION
**Pull request by TigerInYourDream**
Signed-off-by: TigerInYourDreams <TigerInYourDream@hi.zy>

## What have you changed? (mandatory)
Imple the function instr in builtin_string.
Please **NOTE** that:
- Do not assume reviewers understand the original issue

## What are the type of the changes? (mandatory)
implement the function instr which I found in #3275 
The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Misc (other changes)

## How has this PR been tested? (mandatory)
unit test
Please describe the tests that you ran to verify your changes. Have you finished unit tests, integration tests, or manual tests? What additional tests would give you greater confidence in this change?

## Does this PR affect documentation (docs) update? (mandatory)
NO
If there is document change, please file a PR in ([docs](https://github.com/tikv/tikv/tree/master/docs)) and add the PR number here.
NO
## Does this PR affect tidb-ansible update? (mandatory)
NO
If there is a configuration or metrics change, please file a PR in [tidb-ansible](https://github.com/pingcap/tidb-ansible), and add the PR number here.
NO

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_string.rs [hunk 1]
```diff
@@ -3259,4 +3283,43 @@ mod tests {
...
+    #[test]
+    fn test_instr() {
+        let cases: Vec<(&str, &str, i64)> = vec![
+            ("a", "abcdefg", 1),
+            ("0", "abcdefg", 0),
+            ("c", "abcdefg", 3),
+            ("F", "abcdefg", 6),
+            ("cd", "abcdefg", 3),
+            (" ", "abcdefg", 0),
+            ("", "", 0),
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_string.rs]: It seems not correct in mysql 5.6.25

```
mysql> SELECT INSTR('', '');
+---------------+
| INSTR('', '') |
+---------------+
|             1 |
+---------------+
1 row in set (0.00 sec)
```

Comment (TigerInYourDream) [reply src/coprocessor/dag/expr/builtin_string.rs]: Already fixed. Thanks for the review.

Comment (zhangjinpeng1987) [reply src/coprocessor/dag/expr/builtin_string.rs]: Does https://github.com/tikv/tikv/pull/4340 have the same problem?

Comment (TigerInYourDream) [reply src/coprocessor/dag/expr/builtin_string.rs]: @zhangjinpeng1987 I'm sorry. They have the same problem. I didn't test it in mysql. Unfortunately, I delete the origin branch. Should I create a new pull request for it?

Comment (zhangjinpeng1987) [reply src/coprocessor/dag/expr/builtin_string.rs]: Yes, you are welcome.

**Comment by TigerInYourDream** [comment]
@zhangjinpeng1987 @AndreMouche Please take a Look.

File: src/coprocessor/dag/expr/scalar_function.rs [hunk 2]
```diff
@@ -837,6 +837,7 @@ dispatch_call! {
 
         UncompressedLength => uncompressed_length,
         Strcmp => strcmp,
+        Instr=>instr,
```
Comment (rleungx) [inline src/coprocessor/dag/expr/scalar_function.rs]:
```suggestion
        Instr => instr,
```

Comment (TigerInYourDream) [reply src/coprocessor/dag/expr/scalar_function.rs]: > Suggested change 
>         Instr=>instr,
>         Instr => instr,

Already changed. Thanks for the advices

**Comment by rleungx** [comment]
Please resolve the conflicts.

**Comment by TigerInYourDream** [comment]
> Please resolve the conflicts.

Have resolved.

**Comment by iosmanthus** [comment]
It seems that `instr` shares the same logic with `locate_2_args` which I reimplemented in https://github.com/tikv/tikv/pull/4378. Will it cause some conflicts when we merge these two PR? :cry:

**Comment by rleungx** [comment]
@TigerInYourDream Thanks! Merged.

**Pull request merged**